salt/modules/cp.py

```diff
@@ -357,9 +359,11 @@ def get_file_str(path, saltenv='base', env=None):
         saltenv = env
 
     fn_ = cache_file(path, saltenv)
-    with salt.utils.fopen(fn_, 'r') as fp_:
-        data = fp_.read()
-    return data
+    if isinstance(fn_, str):
```

Comment:
This needs to be using `six.string_type` instead of `str` here.

Any reason you're not using `os.path.exists` or something similar instead of checking for a str type?